lib/situation_room/check/check_make.ex

```diff
@@ -0,0 +1,25 @@
+defmodule SituationRoom.Check do
+  @moduledoc """
+  Logic for sending requests to site URLs
+  """
+  alias SituationRoom.RequestClient
+
+  def run(url) do
+    start_ms = :os.system_time(:microsecond)
+    res = RequestClient.head(url)
+    time = :os.system_time(:microsecond) - start_ms
+
+    case res do
+      {:ok, response} ->
+        {:ok, %{"status" => response.status, "response_time" => time / 1000}}
+
+      {:error, {reason}} ->
+        {:error, reason}
+
+      {:error, reason} ->
+        {:error, reason}
+    end
+  rescue
```

Comment:
What is being rescued here and why?  Why the rescue vs graceful handling and/or just let the process die and restart?

Reply:
Prior to this, I thought of rescue as a 'catchall' to fallback on. In hindsight this is a poor way of dealing with an error that I was expecting. When given a bad input for the run function, such as a number or an empty string, the process will die. I suppose a function spec or a guard clause could catch this error as well. I will also look into graceful handling.